### ISSUE TITLE
Train config file

### ISSUE THREAD

**Issue by bulygin1985** [bulygin1985/object_detection#8]
yml or json with all train parameters:
Each component should have Parent class (OOP)
1) backbone
2) head
3) lr schedule
4) loss function with parameters
5) augmentation, preprocessing
6) overfit/train mode (number of images for overfit)

**Comment by msvsn** [comment]
7. num of epochs

**Comment by olnkravchenko** [comment]
JSON example:
`{
  "model": {
    "alpha": 0.25
  },
  "train": {
    "lr_schedule": {
      "patience": 100,
      "factor": 0.5
    },
    "lr": 0.01,
    "mode": "overfit",
    "stop_criteria": {
      "loss": null,
      "max_epoch": 500
    }
  }
}`

**Comment by olnkravchenko** [comment]
CLI example:
`python train.py --config filename.json`

**Comment by olnkravchenko** [comment]
```
import json
parser = argparse.ArgumentParser()
parser.add_argument("-c", "--config", type=str, help="path to config file")
args = parser.parse_args()

filepath = args.config
with open(filepath, 'r') as f:
    config = json.load(f)
```
